src/core/components/collapsible/collapsible.api.ts

```diff
@@ -0,0 +1,39 @@
+import { type ComponentProps } from 'react';
+import { ButtonVariant } from '../button';
+
+export type CollapsedSize = 'sm' | 'md' | 'lg';
+
+export interface ICollapsibleProps extends ComponentProps<'div'> {
+    /**
+     * The initial height of the collapsible container while closed. @default md
+     */
+    collapsedSize?: CollapsedSize;
+    /**
+     * Custom size for the collapsible container that will override collapsedSize prop. See 'Spacing' docs for valid input values.
+     */
+    customCollapsedSize?: number;
+    /**
+     * Default state of the collapsible container. @default false
+     */
+    defaultOpen?: boolean;
+    /**
+     * The label to display on the trigger button when the collapsible container is closed.
+     */
+    buttonLabelClosed?: string;
+    /**
+     * The label to display on the trigger button when the collapsible container is open.
+     */
+    buttonLabelOpened?: string;
+    /**
+     * Override the generic trigger button with an ODS Button component by passing a variant.
+     */
+    buttonVariant?: ButtonVariant;
```

Comment:
I figured that this gave it the most flexibility long term for extending into other ODS components or a slightly more custom user impl.

Reply:
This does mean thought that ButtonVariant is useless because no matter what you pass in, you get tertiary. I would rather make up Ghost & Tertiary like so: 
```ts
CollapsibleTriggerVariant = Pick<ButtonVariant, 'tertiary'> & 'ghost';
```